src/system/dev/vfs.c

```diff
@@ -89,19 +89,27 @@ int vfs_update_entry(int file, struct fs_driver const* const driver, void* arg)
 	return 0;
 }
 
-int _open(const char* file, int flags, int mode) {
-	struct _reent* r = _REENT;
-	// Check if the filename is too long or not NULL terminated
+int file_size(const char* file) {
+	// Returns true if is under size constraint
 	size_t i = 0;
 	for (i = 0; i < MAX_FILELEN; i++) {
 		if (file[i] == '\0') {
 			break;
 		}
 	}
 	if (i == MAX_FILELEN) {
+		return i != MAX_FILELEN;
+	}
```

Comment:
You don't want the surrounding if right?

Reply:
to expand on that: if `i == MAX_FILELEN`, you then return `i != MAX_FILELEN`, which will always be 0 (false). you can remove the if statement entirely, and just have the last line of this function be `return i != MAX_FILELEN`

Reply:
To expand a bit further, since it seams you don't realize what I pointed out originally. This:
`if (condition)`
`return false;`
`else`
    `return true;`

is the exact same as:

`return !(condition);`

It's just that one's a shorter and nicer.

Reply:
All done